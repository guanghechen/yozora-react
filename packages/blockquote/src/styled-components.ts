import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
