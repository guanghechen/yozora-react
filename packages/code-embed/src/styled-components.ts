import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
