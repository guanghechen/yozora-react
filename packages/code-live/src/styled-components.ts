import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
