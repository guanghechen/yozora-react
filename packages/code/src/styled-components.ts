import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
