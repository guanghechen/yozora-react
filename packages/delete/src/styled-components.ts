import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
