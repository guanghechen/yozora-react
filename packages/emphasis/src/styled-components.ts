import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
