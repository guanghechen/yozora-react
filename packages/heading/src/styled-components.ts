import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
