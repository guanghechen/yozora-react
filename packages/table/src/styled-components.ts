import type { YozoraTableTheme } from './theme'
import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    table?: YozoraTableTheme
  }
}
