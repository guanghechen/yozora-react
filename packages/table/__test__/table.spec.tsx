import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Table from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  const rows: React.ReactNode[] = [
    (<tr key="0"><th>Name</th></tr>),
    (<tr key="1"><td>Alice</td></tr>),
    (<tr key="2"><td>Bob</td></tr>),
  ]

  it('render with custom className', () => {
    const className = 'custom-list-item'
    const wrapper = render(
      <Table className={ className } data-value="waw">
        { rows }
      </Table>
    )
    expect(wrapper.find('.' + className) != null).toEqual(true)
    expect(wrapper.text()).toEqual('NameAliceBob')
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<Table>{ value }</Table>)
      }).toThrow(/Failed prop type: the prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLTableElement>()
    const wrapper = mount(
      <Table ref={ ref } data-value="waw">
        { rows }
      </Table>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Table style={{ color: 'orange', fontSize: '16px' }}>
        { rows }
      </Table>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        table: {
          width: '100%',
          overflow: 'hidden auto',
          margin: 18,
        }
      }
    }

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <Table style={{ color: 'orange', fontSize: '16px' }}>
          { rows }
        </Table>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
