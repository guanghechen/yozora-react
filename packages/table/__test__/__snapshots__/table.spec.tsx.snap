// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<table
  class="yozora-table"
  style="color:orange;font-size:16px"
>
  <thead
    class="yozora-table__thead"
  >
    <tr
      class="yozora-table-row"
    >
      <th
        align="center"
        class="yozora-table-cell"
      >
        Name
      </th>
      <th
        align="left"
        class="yozora-table-cell"
      >
        Age
      </th>
    </tr>
  </thead>
  <tbody
    class="yozora-table__tbody"
  >
    <tr
      class="yozora-table-row"
    >
      <td
        align="center"
        class="yozora-table-cell"
      >
        Alice
      </td>
      <td
        align="left"
        class="yozora-table-cell"
      >
        18
      </td>
    </tr>
    <tr
      class="yozora-table-row"
    >
      <td
        align="center"
        class="yozora-table-cell"
      >
        Bob
      </td>
      <td
        align="left"
        class="yozora-table-cell"
      >
        19
      </td>
    </tr>
  </tbody>
</table>
`;
